zingolib/src/lightclient/send.rs

```diff
@@ -159,18 +157,61 @@ impl LightClient {
     #[cfg(feature = "zip317")]
     /// Unstable function to expose the zip317 interface for development
     // TODO: add correct functionality and doc comments / tests
-    pub async fn do_send_proposal(&self) -> Result<Vec<TxId>, String> {
+    async fn do_send_proposal<NoteRef>(
+        &self,
+        _proposal: &Proposal<zcash_primitives::transaction::fees::zip317::FeeRule, NoteRef>,
+    ) -> Result<NonEmpty<TxId>, String> {
+        Ok(NonEmpty::singleton(TxId::from_bytes([222u8; 32])))
+    }
+
+    #[cfg(feature = "zip317")]
+    /// Unstable function to expose the zip317 interface for development
+    // TODO: add correct functionality and doc comments / tests
+    pub async fn do_send_stored_proposal(&self) -> Result<NonEmpty<TxId>, String> {
         if let Some(proposal) = self.latest_proposal.read().await.as_ref() {
             match proposal {
-                crate::lightclient::ZingoProposal::Transfer(_) => {
-                    Ok(vec![TxId::from_bytes([1u8; 32])])
+                crate::lightclient::ZingoProposal::Transfer(transfer_proposal) => {
+                    self.do_send_proposal::<NoteId>(transfer_proposal).await
                 }
-                crate::lightclient::ZingoProposal::Shield(_) => {
-                    Ok(vec![TxId::from_bytes([222u8; 32])])
+                crate::lightclient::ZingoProposal::Shield(shield_proposal) => {
+                    self.do_send_proposal::<Infallible>(shield_proposal).await
                 }
             }
         } else {
             Err("No proposal. Call do_propose first.".to_string())
         }
     }
+
+    #[cfg(feature = "zip317")]
+    /// Unstable function to expose the zip317 interface for development
+    // TODO: add correct functionality and doc comments / tests
+    pub async fn do_quick_send(
+        &self,
+        request: TransactionRequest,
+    ) -> Result<NonEmpty<TxId>, String> {
+        if let Ok(proposal) = self.do_propose_send(request).await {
+            self.do_send_proposal::<NoteId>(&proposal).await
+        } else {
+            Err("No proposal. Call do_propose first.".to_string())
```

Comment:
This is...extremely misleading. This error occurs in the case where the proposal fails to be created, not because the creation wasn't called.